api/tests/src/db/models/factories.py

```diff
@@ -1821,3 +1821,64 @@ class Meta:
     last_notified_at = factory.Faker("date_time_between", start_date="-5y", end_date="-3y")
 
     searched_opportunity_ids = factory.LazyAttribute(lambda _: random.sample(range(1, 1000), 5))
+
+
+class OpportunityCompetitionFactory(BaseFactory):
+    class Meta:
+        model = opportunity_models.OpportunityCompetition
+
+    opportunity = factory.SubFactory(OpportunityFactory)
+    opportunity_id = factory.LazyAttribute(lambda o: o.opportunity.opportunity_id)
+
+    competition_id = Generators.UuidObj
+    legacy_competition_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
+    public_competition_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
+    legacy_package_id = sometimes_none(factory.Faker("random_int", min=1, max=15))
```

Comment:
For these, my recommendation:
* don't set legacy_competition_id -> it's a pointer back to the old system, only our transform logic will ever care about it / want it set.
* public_competition_id -> It's actually a string with some variance in how it can be set, I'd just put something like `ABC<number>` to make it look sorta like the old system (keep the sometimes_none)
* legacy_package_id - This looks to be formatted as `PKG<number>`, probably just do that as well? (keep the sometimes_none).